lib/PayPal/Rest/ApiContext.php

```diff
@@ -85,14 +85,20 @@ public function addRequestHeader($name, $value)
      */
     public function getRequestId()
     {
-        if ($this->requestId == null) {
-            $this->requestId = $this->generateRequestId();
-        }
-
         return $this->requestId;
     }
 
     /**
+     * Sets the request ID
+     *
+     * @param string $requestId the PayPal-Request-Id value to use
+     */
+    public function setRequestId($requestId)
+    {
+        $this->requestId = $requestId;
+    }
+
+    /**
```

Comment:
@jaypatel512 and @sdcoffey Deprecated both `generateRequestId` and `resetRequestId`. We didn't want to delete `resetRequestId` since it is a public method and would be an API change. It also could have been used correctly in the past to reset the request ID between idempotent requests so the behavior is left as-is.

The `generateRequestId` method is still valid but I would recommend a UUID package and calling `setRequestId` with that value. I don't want to add a UUID dependency to this SDK though and wanted to at least keep the old behavior of setting to a relatively unique value inside `resetRequestId` so that's why it wasn't removed (even though it is a private method).